openedx/core/djangoapps/programs/management/commands/backpopulate_program_credentials.py

```diff
@@ -38,13 +40,53 @@ def add_arguments(self, parser):
             default=False,
             help='Submit tasks for processing.'
         )
+        parser.add_argument(
+            '--args-from-database',
+            action='store_true',
+            default=False,
+            help='Use arguments from the Config model instead of the command line.',
+        )
+        parser.add_argument(
+            '--programs',
+            nargs='+',
+            help='Award certificates only for specific programs.',
+        )
+        parser.add_argument(
+            '--users',
+            nargs='+',
+            help='Award certificates only to specific users.',
+        )
+
+    def get_args_from_database(self):
+        """ Returns an options dictionary from the current NotifyCredentialsConfig model. """
+        config = CustomProgramsConfig.current()
+        if not config.enabled:
+            raise CommandError('CustomProgramsConfig is disabled, but --args-from-database was requested.')
+
+        # We don't need fancy shell-style whitespace/quote handling - none of our arguments are complicated
+        argv = config.arguments.split()
+
+        parser = self.create_parser('manage.py', 'backpopulate_program_credentials')
+        return parser.parse_args(argv).__dict__   # we want a dictionary, not a non-iterable Namespace object
 
     def handle(self, *args, **options):
-        logger.info('Loading programs from the catalog.')
-        self._load_course_runs()
+        if options['args_from_database']:
+            logger.info('Loading arguments from the database.')
+            arguments = self.get_args_from_batabase()
+            if arguments['programs']:
```

Comment:
We can simplify it like this:
```
program_uuids = arguments.get('programs', None)
usernames = arguments.get('users', None)
self._load_course_runs(program_uuids=program_uuids)
self._load_usernames(users=usernames)
```

Reply:
Updated the code according to the suggestion.